app/models/QualifierOfIdentification.scala

```diff
@@ -56,6 +58,13 @@ object QualifierOfIdentification extends EnumerableType[QualifierOfIdentificatio
 
   val messageKeyPrefix: String = "qualifierOfIdentification"
 
+  def values(userAnswers: UserAnswers): Seq[QualifierOfIdentification] =
+    userAnswers.get(TypeOfLocationPage) match {
```

Comment:
As a refactor for the controller you could also make 'TypeOfLocationPage' a mandatory page in the actions and just pass the value here. This means you could get rid of the optionality here.